include/zephyr/pm/power_gating.h

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2022 Intel corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_POWER_GATING_H_
+#define ZEPHYR_INCLUDE_POWER_GATING_H_
+
+#include <kernel.h>
+
+/**
+ * @brief Power gating API
+ * @defgroup subsys_pm_pg Power Gating
+ * @ingroup subsys_pm
+ */
+
+/**
+ * @struct power_reg
+ * @brief Structure for specifying address and bits used in power gating
+ *
+ * @param wr_address Address of register used to set PG
+ * @param rd_address Address of register used to read PG status
+ * @param SPA_bit Set Power Active bit in wr_address used to toggle power on/off
+ * @param CPA_bit Confirm Power Active bit in rd_address used to verify power gating state
+ */
+struct pm_pg_registers {
+  uint32_t wr_address;
+	uint32_t rd_address;
+  uint32_t SPA_bit;
+  uint32_t CPA_bit;
```

Comment:
```suggestion
  uint32_t wr_address;
	uint32_t rd_address;
  uint32_t SPA_bit;
  uint32_t CPA_bit;
```
Indentation is wrong, use tabs please